File: src/etch/bindings/java/msg/Validator_double.java
package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for double.
 */
public class Validator_double extends TypeValidator
{
	/**
	 * @param nDims the number of dimensions. 0 for a scalar.
	 * @return an instance of the validator.
	 */
	public static Validator_double get( int nDims )
    {
		if (nDims < 0)
			return null;
		
		if (nDims >= validators.length)
			return new Validator_double( nDims );
		
		if (validators[nDims] == null)
			validators[nDims] = new Validator_double( nDims );
		
		return validators[nDims];
    }
	
	private static Validator_double[] validators = new Validator_double[4];
	
	/**
	 * Constructs the Validator.
	 *
	 * @param nDims the number of dimensions. 0 for a scalar.
	 */
	private Validator_double( int nDims )
	{
		super( Double.class, double.class, nDims, "double["+nDims+"]" );
	}

	@Override
    public Validator elementValidator()
    {
	    return get( nDims-1 );
    }

	@Override
	public Byte checkValue( Object value )
	{
		if (!validate( value))
			return null;
		
		if (nDims > 0)
			return TypeCode.ARRAY;
		
		return TypeCode.FLOAT8;
	}
}

File: src/etch/bindings/csharp/msg/Validator_double.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for double
    /// </summary>
    public class Validator_double : TypeValidator
    {
        /// <summary></summary>
        /// <param name="nDims">number of dimensions. 0 for a scalar.</param>
        /// <returns>an instance of the validator</returns>
        public static Validator_double Get( int nDims )
        {
            CheckDims( nDims );

            if ( nDims >= validators.Length )
                return new Validator_double( nDims );

            Validator_double v = validators[ nDims ];

            if ( v == null )
                v = validators[ nDims ] = new Validator_double( nDims );

            return v;
        }

        private static Validator_double[] validators = new Validator_double[ MAX_CACHED ];

        /// <summary>
        /// Constructs the validator.
        /// </summary>
        /// <param name="nDims"></param>
        private Validator_double( int nDims )
            : base( typeof( Double ), typeof( double ), nDims, "double["+nDims+"]" )
        { 
        }

        public override Validator ElementValidator()
        {
            return Get( nDims - 1 );
        }
       
    }
}
